xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/migration/R14010XWIKI19352DataMigration.java

```diff
@@ -0,0 +1,143 @@
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.index.migration;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.index.TaskManager;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+import org.xwiki.wiki.descriptor.WikiDescriptorManager;
+import org.xwiki.wiki.manager.WikiManagerException;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.store.migration.DataMigrationException;
+import com.xpn.xwiki.store.migration.XWikiDBVersion;
+import com.xpn.xwiki.store.migration.hibernate.HibernateDataMigration;
+
+import static org.xwiki.index.internal.DefaultLinksTaskConsumer.LINKS_TASK_TYPE;
+
+/**
+ * Migrates the links by reindexing all the documents of the farm.
+ *
+ * @version $Id$
+ * @since 14.2RC1
+ */
+// TODO: Implement DataMigration once XWIKI-19399 is fixed.
+@Component
+@Singleton
+@Named(R14010XWIKI19352DataMigration.HINT)
+public class R14010XWIKI19352DataMigration implements HibernateDataMigration
+{
+    /**
+     * The hint for this component.
+     */
+    public static final String HINT = "R14010XWIKI19352";
+
+    @Inject
+    private WikiDescriptorManager wikiDescriptorManager;
+
+    @Inject
+    private QueryManager queryManager;
+
+    @Inject
+    private TaskManager taskManager;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Override
+    public String getName()
+    {
+        return HINT;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Queue all the document of the wiki for links indexing.";
+    }
+
+    @Override
+    public XWikiDBVersion getVersion()
+    {
+        return new XWikiDBVersion(140100000);
+    }
+
+    @Override
+    public void migrate() throws DataMigrationException
+    {
+        XWikiContext context = this.contextProvider.get();
+
+        try {
+            for (String wikiId : this.wikiDescriptorManager.getAllIds()) {
+                context.setWikiId(wikiId);
```

Comment:
Hmmm, @tmortagne would know better than me, but if I'm correct the migration code is triggered independently on each wiki: you shouldn't need to loop yourself on the DB.

Reply:
Yes, each wiki has its own migration process, so a migration is supposed to only deal with the database.